python/lsst/pipe/tasks/measurePsf.py

```diff
@@ -278,12 +277,15 @@ def run(self, exposure, sources, expId=0, matches=None):
         reserveList = []
 
         if self.config.reserveFraction > 0:
-            random.seed(self.config.reserveSeed*expId)
-            reserveList = random.sample(psfCandidateList,
-                                        int((self.config.reserveFraction)*len(psfCandidateList)))
-
-            for cand in reserveList:
-                psfCandidateList.remove(cand)
+            random = afwMath.Random(seed = self.config.reserveSeed*expId)
+            reserveList = []
+            n = len(psfCandidateList)
+            for i in range(int(n * self.config.reserveFraction)):
```

Comment:
See https://developer.lsst.io/coding/python_style_guide.html#binary-operators-should-be-surrounded-by-a-single-space-except-for